damnit/ctxsupport/damnit_ctx.py

```diff
@@ -128,6 +157,301 @@ def evaluate(self, run_data, kwargs):
 
         return cell
 
+    def copy(self):
+        return copy(self)
+
+
+def is_group(obj) -> bool:
+    """Return True if obj is a Group class or an instance of a Group class."""
+    cls = obj if isinstance(obj, type) else type(obj)
+    return hasattr(cls, GROUP_MARKER_ATTR)
+
+
+def is_group_instance(obj) -> bool:
+    """Return True if obj is an instance of a Group class.
+    """
+    return hasattr(type(obj), GROUP_MARKER_ATTR)
+
+
+class Group:
+    """A class decorator that transforms a class into a Group.
+
+    Transforms a class into a Group 'dataclass'. Allow for creating reusable,
+    configurable, and nestable sets of related Variables.
+
+    - `@Group` for default behavior.
+    - `@Group(title='...', ...)` to provide parameters.
+
+    Args:
+        title (str, optional): The title of the group.
+        tags (list[str], optional): A list of tags to associate with the group.
+        sep (str, optional): Separator to use between group title and variable title.
+    """
+    # _REGISTRY: WeakValueDictionary[str, 'Group'] = WeakValueDictionary()
+
+    def __new__(cls, decorated_cls: type = None, **kwargs) -> type:
```

Comment:
Maybe replace **kwargs here with the explicit keyword args for the GroupBase fields, and then get rid of the check on keys passed in.